packages/stateless/components/dao/tabs/AppsTab.tsx

```diff
@@ -63,25 +63,27 @@ const InnerAppsTab = ({
 }: InnerAppsTabProps) => {
   const { t } = useTranslation()
   const [iframe, setIframe] = useState<HTMLIFrameElement | null>(null)
+  const [inputUrl, setInputUrl] = useState<string>(url)
 
-  const go = (destUrl = url) => {
-    setUrl(destUrl)
-    if (iframe) {
-      iframe.src = destUrl
-    }
+  const go = (url: string) => {
+    setUrl(url)
   }
 
-  // On first iframe mount, go to url if valid already.
+  // On first iframe mount, go to url
   useEffect(() => {
     try {
-      if (iframe && url && new URL(url).href) {
+      if (iframe) {
         iframe.src = url
       }
     } catch {
       // Ignore.
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [iframe])
+  }, [iframe, url])
+
+  useEffect(() => {
+    if(url !== inputUrl)
+      setInputUrl(url);
+  }, [setInputUrl, url])
```

Comment:
```suggestion
  // Update the input field to match the URL if it changes in the parent
  // component. This should handle the URL being updated from the query params.
  useEffect(() => {
    if (url !== inputUrl) {
      setInputUrl(url)
    }
    // Only change the input URL when the URL changes (i.e. ignore input change).
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [setInputUrl, url])
```

looks good but the formatter is going to complain. running `yarn format` before pushing commits should solve this generally.